maven_plugin/src/main/java/io/github/chains_project/maven_lockfile/graph/DependencyNode.java

```diff
@@ -35,25 +31,14 @@ public class DependencyNode implements Comparable<DependencyNode> {
 
     private final Set<DependencyNode> children;
 
-    // kept for backward compability, following https://github.com/chains-project/maven-lockfile/pull/803
```

Comment:
Somewhat surprised by this comment, unless the constructor was originally kept to require less refactoring.

Removing this should be safe as the constructor is package-local, and not part of any public API.